src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,207 @@
+
+    public String getCommand() {
+        return parseInput()[0];
+    }
+
+    /**
+     *  Return index after parsing the input like for mark/unmark index based on user
+     *
+     * @return the index has been registered
+     */
+    public String getIndex() {
+        return parseInput()[1];
+    }
+
+    public Task processEvent() {
+        if(inputArray.length <= 2) {
+            return null;
+        }
+
+
+        Integer startIndex = -1;
+        String extractedTask = String.join(" ", Arrays.copyOfRange(inputArray, 1, inputArray.length));
+
+        String endDate = "";
+        String startDate = "";
+        for(int i = 0; i < inputArray.length; i++) {
+            if(inputArray[i].equals("/from") && startIndex == -1) {
+                startIndex = i;
+                extractedTask = String.join(" ", Arrays.copyOfRange(inputArray, 1, i ));
+            } else if(inputArray[i].equals("/to") && startIndex != -1) {
+                endDate = String.join(" ", Arrays.copyOfRange(inputArray, i+1, inputArray.length));
+                startDate = String.join(" ", Arrays.copyOfRange(inputArray, startIndex + 1, i));
+            }
+        }
+
+        Task newTask = new Event(extractedTask,convertDateTime(startDate),convertDateTime(endDate));
+        return newTask;
+
+    }
+
+        //Using format given by textbook dd-mm-yyyy HHmm example :"02/12/2019 1800"
+    /**
+     * Convert the LocalDateTime based on String formatted input
+     *
+     * @param input the input has to be format of dd-mm-yyyy HHmm, example : "02/12/2019 1800"
+     * @return The "translated" datetime based on input
+     */
+    public LocalDateTime convertDateTime(String input) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+        LocalDateTime dateTime = LocalDateTime.parse(input, formatter);
+
+        return dateTime;
+    }
+
+
+    /**
+     * ProcessDeadline based on the userinput, slicing and categorise with text, dueDate
+     * then create and return a deadline task
+     *
+     * @return Deadline Task
+     */
+    public Task processDeadline() {
+        if(inputArray.length <= 2) {
+            return null;
+        }
+
+        String dueDate = "";
+        String extractedTask = String.join(" ", Arrays.copyOfRange(inputArray, 1, inputArray.length));
+
+        for(int i = 0; i < inputArray.length; i++) {
+            if(inputArray[i].equals("/by")) {
+                dueDate = String.join(" ", Arrays.copyOfRange(inputArray, i+1, inputArray.length));
+                extractedTask = String.join(" ", Arrays.copyOfRange(inputArray, 1, i));
+
+                break;
+            }
+        }
+
+        Task newTask = new Deadline(extractedTask,convertDateTime(dueDate));
+        return newTask;
+    }
+
+    /**
+     * ProcessToDo based on the userinput, slicing, save and create todo task
+     *
+     * @return Todo Task
+     */
+    public Task processToDo() {
+        if(inputArray.length <= 1) {
```

Comment:
I believe you need a whitespace here: something like
`if (inputArray.length <= 1) { `

Reply:
![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/94635140/addbddb4-df39-4bb3-a4db-46bfa9cf2a45)

If you wanted to see where it's from you can also go [here](https://se-education.org/guides/conventions/java/intermediate.html)